homeassistant/components/sensor/qbittorrent.py

```diff
@@ -0,0 +1,140 @@
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+async def async_setup_platform(hass, config, add_entities,
+                               discovery_info=None):
+    """Set up the qbittorrent sensors."""
+    from qbittorrent.client import Client, LoginRequired
+
+    try:
+        client = Client(config.get(CONF_URL))
+        client.login(config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
+    except LoginRequired:
+        _LOGGER.info("Invalid authentication for qBittorrent. Check config.")
+        return
+    except RequestException:
+        _LOGGER.error("Connection to qBittorrent failed. Check config.")
+        raise PlatformNotReady
+
+    name = config.get(CONF_NAME)
+
+    dev = []
+    for sensor_type in SENSOR_TYPES:
+        sensor = QBittorrentSensor(sensor_type, client, name, LoginRequired)
+        dev.append(sensor)
+        await sensor.async_update()
+
+    add_entities(dev)
+
+
+def format_speed(speed):
+    """Return a bytes/s measurement as a human readable string."""
+    kb_spd = float(speed) / 1024
+    return round(kb_spd, 2 if kb_spd < 0.1 else 1)
+
+
+class QBittorrentSensor(Entity):
+    """Representation of an qbittorrent sensor."""
+
+    def __init__(self, sensor_type, qbittorrent_client,
+                 client_name, exception):
+        """Initialize the sensor."""
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self.client = qbittorrent_client
+        self.type = sensor_type
+        self.client_name = client_name
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._available = False
+        self._exception = exception
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return true if device is available."""
+        return self._available
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    async def async_update(self):
+        """Get the latest data from qbittorrent and updates the state."""
+        try:
+            data = self.client.sync()
+            self._available = True
+        except RequestException:
+            _LOGGER.error("Connection to qBittorrent lost.")
+            self._available = False
+            return
+        except self._exception:
+            _LOGGER.info("Invalid authentication for qBittorrent."
+                         " Check config.")
+            return
+
+        download = data['server_state']['dl_info_speed']
+        upload = data['server_state']['up_info_speed']
+
+        if self.type == SENSOR_TYPE_CURRENT_STATUS:
+            if data:
```

Comment:
Can be removed. If `data` is `None` or empty there would be an error while getting `download` and `upload`. 

Add a guard clause after the retrieval of the data.

```python
if data is None:
    return
```